### PR TITLE
Added `#include <cstdint>` to some files

### PR DESCRIPTION
**Pull request by ifndefJOSH**
Added `#include <cstdint>` to the following files:

- src/storm/logic/RewardAccumulation.h
- src/storm/modelchecker/prctl/helper/rewardbounded/EpochManager.h
- src/storm/utility/NumberTraits.h
- src/storm/storage/jani/traverser/InformationCollector.h
- src/storm/utility/string.h
- src/storm-cli-utilities/cli.h

I am not sure why, but the omission of this include caused errors on the latest versions of both GCC and Clang on Artix Linux.

Tested with the following C++ compilers
gcc (GCC) 13.1.1 20230429
clang version 15.0.7

CMake version:
cmake version 3.26.3

### CODE REVIEW

File: src/storm-gspn/builder/ExplicitGspnModelBuilder.cpp
Comment (tquatmann) [inline src/storm-gspn/builder/ExplicitGspnModelBuilder.cpp:1]: These seem to be only whitespace changes. Please revert.

File: src/storm-gspn/builder/ExplicitGspnModelBuilder.h
Comment (tquatmann) [inline src/storm-gspn/builder/ExplicitGspnModelBuilder.h:1]: These seem to be only whitespace changes. Please revert.

File: src/storm/exceptions/ExceptionMacros.h
Comment (tquatmann) [inline src/storm/exceptions/ExceptionMacros.h:1]: These seem to be only whitespace changes. Please revert.

**Review by tquatmann — CHANGES_REQUESTED** [review]
Thanks for spotting these!
I can see why these includes could be missing. Could you take care of the whitespace changes I mentioned? Also, the formatting seems to be weird. You can do `make format` (assuming that your system has `clang-format`) to fix that.

**Comment by tquatmann** [comment]
Ohh, my bad! Thanks for clarifying @volkm and sorry for the confusion @ifndefJOSH 

On my machine with `clang-format 16.0.3` I don't get any changes for  `../src/storm/exceptions/ExceptionMacros.h` though.

**Review by tquatmann — APPROVED** [review]

**Comment by tquatmann** [comment]
I took the liberty to revert the format changes to `ExceptionMacros`. Apparently, clang-format 16 (current release) does not produce these changes.
I think this is now ready to be merged, thanks @ifndefJOSH

**Comment by flyingkukri** [comment]
Hi, why don't you push this commit to stable (or maybe add this to the troubleshooting page)?

Because on Fedora 38 building storm without these changes doesn't work.

But still thanks for fixing!

**Comment by volkm** [comment]
Good timing, because we actually released the latest [version 1.8.0](https://github.com/moves-rwth/storm/releases/tag/1.8.0) of Storm today. The `stable` branch is also up-to-date now and contains this fix.

If you encounter any issues, feel free to open a new issue.

**Pull request merged**